src/operator/softmax_output-inl.h

```diff
@@ -43,6 +46,14 @@ struct SoftmaxOutputParam : public dmlc::Parameter<SoftmaxOutputParam> {
     DMLC_DECLARE_FIELD(use_ignore).set_default(false)
     .describe("If set to true, the ignore_label value will not contribute "
       "to the backward gradient");
+    DMLC_DECLARE_FIELD(normalization)
```

Comment:
I don't think this is necessary since 'valid' should automatically be used when using ignore while 'batch' is always done by optimizer.